packages/dds/tree/docs/wip/TreeStorage.md

```diff
@@ -0,0 +1,160 @@
+    This can however require storing extra data in interior nodes when moving children between parents and/or regenerating the children to to account for the parent providing a different portion of the path, when just doing operations like splitting nodes for balancing.
+- Model sequence indexes such that inserts usually don't have to change the indexes of other nodes in the sequence (ex: allow inserting at 2.5, instead of moving everything above 3 over one and inserting at 3). As long as the keys still sort in the right order, it should work fine, but can result in keys getting longer. May involve occasionally normalizing the indexes in a sequence to shorten indexes where inserts keep happening.
+- Subtrees could be used as values in the tree instead of just the leaf values, allowing some more opportunity for schema based compression, or other compression schemes.
+
+TODO: this section needs a better understanding of how to avoid storing full paths all over while allowing tree balancing, inserting which changes sequence indexes, and moves without updating the paths within the moved subtrees.
+This is all interrelated, and I don't have a great solution fully in my head: I may be over or under estimating the complexity and efficiency of a solution.
+
+### Node Identifier B-Tree
+
+Give all nodes a unique identifier, and refer to them using this identifier.
+This is the approach used by experimental shared-tree.
+Can be optimized using id-compression and the chunking scheme prototyped in https://github.com/CraigMacomber/sequence.
+
+This approach does poorly for sequences that have been reordered since that forces the B-Tree to have more entries due to not having chunks which cover contiguous ID space.
+
+This chunking approach can avoid a lot of the indirection costs, and sequential id allocation can help keep access in the B tree using nearby keys a lot of the time,
+but in general it can end up pulling down lots of B-nodes when trees fail to have well clustered IDs, which can happen due to editing.
+Due to this overhead, this design isn't considered in the sceneries below, and is mainly included for comparison.
+This design works decently for in memory storage, but doesn't page in/out data very well since a lot of the data in the pages will be data thats not needed for the part of the tree thats being used.
+
+## Example Sceneries
+
+### Deep Tree
+
+Consider a tree with N nodes where its depth is O(N).
+
+This makes the path to the leaves, and thus the size of the key in the Path B-Tree O(N).
+
+TODO: what would this means for the Path B-Tree? In practice this has to account for how it optimizes storage of paths.
+
+In the logical tree case, its O(N) space, and assuming use of some indirection, O(N) log (N) time to visit the whole tree.
+Inlining can be used to get good seizes for the blobs, and indirection can be used occasionally to lower its costs by a constant factor.
```

Comment:
Nit: sizes

Modeling a linked list might a good motivating example for deep trees?